rag_experiment_accelerator/evaluation/plain_metrics.py

```diff
@@ -1,153 +1,163 @@
 import evaluate
-import textdistance
-from fuzzywuzzy import fuzz
-
-algorithms = textdistance.algorithms
+from rapidfuzz import fuzz
+from rapidfuzz import distance as d
+from textdistance import algorithms as alg
 
 
 # https://huggingface.co/spaces/evaluate-metric/bleu
-def bleu(predictions, references):
-    bleu = evaluate.load("bleu")
-
-    results = bleu.compute(predictions=predictions, references=references, max_order=2)
-    # multiplying by 100 to maintain consistency with previous implementation
-    return results["bleu"] * 100
-
-
-def fuzzy(doc1, doc2):
+def bleu(predictions: list[str], references: list[str]) -> float:
     """
-    Calculates the fuzzy score between two documents.
+    Computes the BLEU score between a list of candidate translations and a list of reference translations.
 
-    Parameters:
-        doc1 (str): The first document to compare.
-        doc2 (str): The second document to compare.
+    Args:
+        predictions (list): A list of candidate translations.
+        references (list): A list of reference translations.
 
     Returns:
-        int: The fuzzy score between the two documents.
+        float: The BLEU score between the candidate and reference translations.
     """
-    differences = fuzzy_compare_values(doc1, doc2)
+    bleu = evaluate.load("bleu")
 
-    return int(sum(differences) / len(differences))
+    results = bleu.compute(predictions=predictions, references=references, max_order=2)
+    # multiplying by 100 to maintain consistency with previous implementation
```

Comment:
Severity: Potential Issue - Consider adding input validation for all functions to ensure the inputs are of the expected types and formats. For example, check that 'predictions' and 'references' are lists of strings in the 'bleu' function.


[Hack - peer-review-pal] We are sorry for the intrusion. Feel free to ignore/delete the comments with the tag - they are automatically generated by [our Hack project](https://github.com/microsoft/peer-review-pal).
If you find them useful, want us to delete them or have any feedback or want to contribute - reach out to some of us: juliame@microsoft.com vadimkirilin@microsoft.com oteslenko@microsoft.com vimeena@microsoft.com
You can prevent future comments from appearing by adding #prp-ignore to your PR description.

Reply:
done